### ISSUE TITLE
devenv.exe

### ISSUE THREAD

**Issue by sj501** [Microsoft/PTVS#2939]
[Window Title]
devenv.exe

[Main Instruction]
An unexpected error occurred

[Content]
Please press Ctrl+C to copy the contents of this dialog and report this error to our issue tracker.

[^] Hide details  [Close]

[Expanded Information]
```
Build: 3.0.17208.1

Microsoft.PythonTools.MissingInterpreterException: The environment "Python 3.6" appears to be incorrectly configured or missing. You may need to install Python 3.6 or re-create a virtual environment.
   at Microsoft.PythonTools.Project.PythonProjectNode.GetInterpreterFactoryOrThrow()
   at Microsoft.PythonTools.Project.PythonProjectNode.GetLaunchConfigurationOrThrow()
   at Microsoft.PythonTools.Project.DefaultLauncherProvider.CreateLauncher(IPythonProject project)
   at Microsoft.PythonTools.PythonToolsPackage.<>c__DisplayClass8_0.<GetLauncher>b__2()
   at Microsoft.VisualStudioTools.UIThread.Invoke[T](Func`1 func)
   at Microsoft.PythonTools.PythonToolsPackage.GetLauncher(IServiceProvider serviceProvider, IPythonProject project)
   at Microsoft.PythonTools.PythonToolsPackage.LaunchFile(IServiceProvider provider, String filename, Boolean debug, Boolean saveDirtyFiles)
   at Microsoft.PythonTools.Language.EditFilter.ExecWorker(Guid& pguidCmdGroup, UInt32 nCmdID, UInt32 nCmdexecopt, IntPtr pvaIn, IntPtr pvaOut)
   at Microsoft.PythonTools.Language.EditFilter.Exec(Guid& pguidCmdGroup, UInt32 nCmdID, UInt32 nCmdexecopt, IntPtr pvaIn, IntPtr pvaOut)
```

**Comment by zooba** [comment]
Could you open Tools, Python, Diagnostic Info and either post the contents as a [gist](https://gist.github.com) and link it here, or email as an attachment to ptvshelp@microsoft.com?

**Comment by zooba** [comment]
Hi @sj501 - we got your email (it came in over a weekend, which is why we didn't respond sooner).

I don't see any environments in your log matching "Python 3.6", which makes me think that the project you tried to debug had some old settings in it? Is that why you closed the issue?

If so, we want to handle the error better, so please let me know and I'll reopen it so we can improve the experience in this case.

**Comment by sj501** [comment]
Hi!
I received an update in VSCode.  Once I updated it (VSCode) my issue was resolved.  I also was experiencing the same issue in VS 2017, both have been resolved or at least I’m no longer receiving the errors on py files.

Issue arose immediately after updating VSCode and VS 2017 respectively.  After receiving the second update this week for VSCode, the issue no longer appears on both platforms.

The issue was across three different laptops I use.  Each issue occurred after the initial update.  In the interim I was using pyCharm as it never experienced the issue.

Thank you for reaching out and checking on me!

Jeff
